### DISCUSSION TITLE
Opening nvim-tree without stealing focus on nvim start

### DISCUSSION THREAD

**Discussion by nydragon** [nvim-tree/nvim-tree.lua#2804] [Q&A] [answered]
Hey, I am trying to get nvim-tree to open on start without stealing focus. I have the following function:

```lua
vim.api.nvim_create_autocmd({ "BufReadPost" }, {
    callback = function()
        if not api.tree.is_visible() then
            api.tree.toggle({
                focus = false,
            })
        end
    end,
})
```

It steals the focus on opening the tree. However, running the following command manually works:

`:lua require("nvim-tree.api").tree.toggle({ focus = false })`

Does anyone have an idea what could be the issue or have a solution/alternative? That would be greatly appreciated!

----

nvim-tree version: 2086e564c4d23fea714e8a6d63b881e551af2f41

[The entire nvim config](https://github.com/Nydragon/nvim)
<details><summary>Lazy Config</summary>
<p>

```lua
"nvim-tree/nvim-tree.lua",
dependencies = {
	"nvim-tree/nvim-web-devicons", -- optional, for file icons
},
init = function()
	require("nvim-tree-config")
end,
opts = {
	renderer = {
		highlight_modified = "name",
	},
	modified = {
		enable = true,
	},
	diagnostics = {
		enable = true,
	},
	update_focused_file = { enable = true },
	filters = {
		dotfiles = false,
		git_ignored = false,
	},
},
},
```

</p>
</details>

**Comment by alex-courtis** [comment, marked as answer]
Sorry, lazy's startup ordering is only somewhat deterministic: wiki: [Lazy Loading](https://github.com/nvim-tree/nvim-tree.lua/wiki/Installation#lazy-loading)

I suggest you take a look at wiki: [Open At Startup](https://github.com/nvim-tree/nvim-tree.lua/wiki/Open-At-Startup) for robust solutions.

**Reply by nydragon** [reply]
Thank you, worked like a charm! My bad for having missed this

**Reply by aep** [reply]
Thanks! Followed this issue because I too didnt find that wiki..
